lib/runtime-core/src/vm.rs

```diff
@@ -14,25 +14,25 @@ use std::{ffi::c_void, mem, ptr};
 #[repr(C)]
 pub struct Ctx {
     /// A pointer to an array of locally-defined memories, indexed by `MemoryIndex`.
-    pub(crate) memories: *mut *mut LocalMemory,
+    pub memories: *mut *mut LocalMemory,
```

Comment:
Is it necessary for these to be public outside of the crate?

Reply:
These fields need to be public to be accessed from the codegen backend. Otherwise we have to access these fields by offsets which is not clean.

Reply:
The ctx has been designed to be usable with falling back on rust. Can you use either offsets or implement it in assembly so that it doesn't need to call rust to do call_indirect?

Reply:
Dynasm provides a quite good feature that computes offset of a struct field automatically by name. Even only for this reason, it's good to keep these fields public.

Also, I prefer to implement cold or large code paths in Rust to improve cleanness and make it less error-prone. Currently the only two functions implemented in assembly are `CONSTRUCT_STACK_AND_CALL_NATIVE` and `CALL_WASM` which both does some low-level stuff that converts calling conventions.

Reply:
The Context is exposed to the user of the runtime, so it's not ideal for these internal fields to be public. I'd be okay with making an `InternalVmCtx` struct that has these fields public, and then is part of the `Vmctx` struct.